e2e_test/js-tests/test_viem_smoketest.mjs

```diff
@@ -0,0 +1,170 @@
+import { assert } from "chai";
+import "mocha";
+import {
+	createPublicClient,
+	createWalletClient,
+	http,
+	defineChain,
+} from "viem";
+import { celoAlfajores } from "viem/chains";
+import { privateKeyToAccount } from "viem/accounts";
+import fs from "fs";
+
+// Load compiled contract
+const testContractJSON = JSON.parse(fs.readFileSync(process.env.COMPILED_TEST_CONTRACT, 'utf8'));
+
+
+// Setup up chain
+const devChain = defineChain({
+	...celoAlfajores,
+	id: 1337,
+	name: "local dev chain",
+	network: "dev",
+	rpcUrls: {
+		default: {
+			http: [process.env.ETH_RPC_URL],
+		},
+	},
+});
+
+const chain = (() => {
+	switch (process.env.NETWORK) {
+		case 'alfajores':
+			return celoAlfajores
+		default:
+			return devChain
+	};
+})();
+
+// Set up clients/wallet
+const publicClient = createPublicClient({
+	chain: chain,
+	transport: http(),
+});
+const account = privateKeyToAccount(process.env.ACC_PRIVKEY);
+const walletClient = createWalletClient({
+	account,
+	chain: chain,
+	transport: http(),
+});
+
+// check checks that the receipt has status success and that the transaction
+// type matches the expected type, since viem sometimes mangles the type when
+// building txs.
+async function check(txHash, type) {
+	const receipt = await publicClient.waitForTransactionReceipt({ hash: txHash });
+	const transaction = await publicClient.getTransaction({ hash: txHash });
+	assert.equal(transaction.type, type, "transaction type does not match");
+	assert.equal(receipt.status, "success", "receipt status 'failure'");
+}
+
+// sendTypedTransaction sends a transaction with the given type and an optional
+// feeCurrency.
+async function sendTypedTransaction(type, feeCurrency){
+		 return await walletClient.sendTransaction({
+			to: "0x00000000000000000000000000000000DeaDBeef",
+			value: 1,
+			type: type,
+			feeCurrency: feeCurrency,
+		});
+}
+
+// sendTypedCreateTransaction sends a create transaction with the given type
+// and an optional feeCurrency.
+async function sendTypedCreateTransaction(type, feeCurrency){
+		 return await walletClient.deployContract({
+			type: type,
+			feeCurrency: feeCurrency,
+			bytecode: testContractJSON.bytecode.object,
+			abi: testContractJSON.abi,
+			 // The constructor args for the test contract at ../debug-fee-currency/DebugFeeCurrency.sol
+			args:[1n, true, true, true],
+		});
+}
+
+// verifyTypedTransactions is a helper function that submits a send and create
+// transaction of the given type with optional feeCurrency and checks the
+// results.
+async function verifyTypedTransactions(type, feeCurrency){
+		const send = await sendTypedTransaction(type, feeCurrency);
+		await check(send, type);
+		const create = await sendTypedCreateTransaction(type, feeCurrency);
+		await check(create, type);
+}
+
+describe("viem smoke test", () => {
+	it("send legacy tx", async () => {
+		const type = "legacy";
+		await verifyTypedTransactions(type);
+	}).timeout(30_000);
```

Comment:
Is there a reason why this timeout is larger than the others?

Reply:
No, updated!